### PR TITLE
chore(deps): prepare for Protobuf v22.x

### LINKED ISSUE #11276: Prepare for Protobuf v22.x
Protobuf v22 has breaking changes affecting the generator (or at least its tests).

```
Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
In file included from ./generator/testing/descriptor_pool_fixture.h:18,
                 from generator/internal/format_method_comments_test.cc:16:
./generator/testing/fake_source_tree.h:34:46: error: 'google::protobuf::io::ZeroCopyInputStream* google::cloud::generator_testing::FakeSourceTree::Open(const std::string&)' marked 'override', but does not override
   34 |   google::protobuf::io::ZeroCopyInputStream* Open(
      |                                              ^~~~
./generator/testing/descriptor_pool_fixture.h:76:18: error: cannot declare field 'google::cloud::generator_testing::DescriptorPoolFixture::source_tree_' to be of abstract type 'google::cloud::generator_testing::FakeSourceTree'
   76 |   FakeSourceTree source_tree_;
      |                  ^~~~~~~~~~~~
./generator/testing/fake_source_tree.h:27:7: note:   because the following virtual functions are pure within 'google::cloud::generator_testing::FakeSourceTree':
   27 | class FakeSourceTree : public google::protobuf::compiler::SourceTree {
      |       ^~~~~~~~~~~~~~
In file included from ./generator/testing/fake_source_tree.h:18:
bazel-out/k8-fastbuild/bin/external/com_google_protobuf/src/google/protobuf/compiler/_virtual_includes/importer/google/protobuf/compiler/importer.h:248:36: note:     'virtual google::protobuf::io::ZeroCopyInputStream* google::protobuf::compiler::SourceTree::Open(absl::lts_20230125::string_view)'
  248 |   virtual io::ZeroCopyInputStream* Open(absl::string_view filename) = 0;
      |                                    ^~~~
```

Note the change from `std::string const&` to `abs::string_view`.

### PR DESCRIPTION
**Pull request by coryan**
Some changes needed because Protobuf v22.x is a major release for C++, with breaking changes.

Fixes #11276

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11278)

### CODE REVIEW

File: generator/generator_test.cc [hunk 1]
```diff
@@ -15,7 +15,9 @@
 #include "generator/generator.h"
 #include "generator/internal/printer.h"
 #include "generator/testing/error_collectors.h"
+#include "generator/testing/fake_source_tree.h"
 #include "generator/testing/printer_mocks.h"
+#include "google/cloud/internal/type_traits.h"
```
Comment (devbww) [inline generator/generator_test.cc:20]: Q.  Was this addition required?

File: generator/testing/fake_source_tree.cc [hunk 2]
```diff
@@ -26,8 +26,8 @@ void FakeSourceTree::Insert(std::string const& filename, std::string contents) {
 }
 
 google::protobuf::io::ZeroCopyInputStream* FakeSourceTree::Open(
-    std::string const& filename) {
-  auto iter = files_.find(filename);
+    FilenameType filename) {
+  auto iter = files_.find(std::string(filename));
```
Comment (devbww) [inline generator/testing/fake_source_tree.cc:30]: Observation:  It's a shame we make a copy (particularly in a "zero copy" function :-), but no biggie.

**Review by devbww — APPROVED** [review]

File: generator/generator_test.cc (hunk 1 above)
Comment (coryan) [reply generator/generator_test.cc]: Removed. Thanks for noticing.

File: generator/testing/fake_source_tree.cc (hunk 2 above)
Comment (coryan) [reply generator/testing/fake_source_tree.cc:30]: Indeed.
